### PR TITLE
Add support for custom proxy credentials using the same policy as adopted by Paket

### PR DESCRIPTION
**Pull request by ruhullahshah**
- Use the same policy as adopted by Paket for proxy handling. In case there is a need to declare a custom proxy, use HTTP_PROXY/HTTPS_PROXY in the command line.

- Code for proxy handling and memoization taken from Paket

Aims to fix https://github.com/fsharp/FAKE/issues/1660

### CODE REVIEW

File: src/app/FakeLib/Utils.fs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿[<AutoOpen>]
```
Comment (matthid) [inline src/app/FakeLib/Utils.fs]: We try to not do this anymore.

**Comment by matthid** [comment]
Thanks looks good, only the `AutoOpen` should be removed.

**Comment by ruhullahshah** [comment]
You are welcome. I have updated the PR to address the review comments.

**Comment by matthid** [comment]
Thanks

**Comment by ruhullahshah** [comment]
You are welcome :)

**Pull request merged**